### PR TITLE
Feat: dockerize development (gitcoin)

### PR DESCRIPTION
**Pull request by JeremiGendron**
## Proposed changes

Mongo container listening on port 27017 (default), pulled from mongo docker hub directly.
Purge script to remove all elements created by the docker-compose

#28 Dockerize the App

This is a docker solution for the mongo portion of the stack, it is possible to mount the node project folder to another container and have that one communicate with the other but there isn't much of an advantage to do so.

It responds to requests on the appropriate port.  With this container, there is no need to install mongo locally; all data is persisted in ${docker_volumes_root}/mongo

Unable to test on my end, getting jwt malformed error, I invite you to test these with your own understanding of the project to confirm whether it works well or not.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/ColuLocalNetwork/inventory-manager/blob/master/.github/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "The inventory manager is a server implementation for community currency issuers",
   "main": "index.js",
   "scripts": {
-    "start": "node index.js",
+    "start": "node index.js --ENV PRODUCTION",
```
Comment (LiorRabin) [inline package.json]: @JeremiGendron actually I want this part to stay without the `ENV` variable

**Comment by LiorRabin** [comment]
Hello @JeremiGendron, I don't quite understand - have you finished the mongo docker part and waiting fro me to approve you on gitcoin to continue dockerizing the app itself?

**Comment by JeremiGendron** [comment]
No, unless you absolutely want me to. There's no point in dockerizing it
 considering you most likely want npm on your system. If you want an npm
container I'll gladly add it, but it won't be as intuitive.

It's easy to have npm/docker and spin up a mongo container in the
background. It's not as convenient to have to attach to a container to
develop and test the app.

**Comment by LiorRabin** [comment]
@JeremiGendron I prefer to have both dockerized, so people can use the app without setting up the entire npm environment on their local machine. Also I would like for people to be able to use docker on their qa/prod environments.

**Comment by JeremiGendron** [comment]
On it!

**Comment by JeremiGendron** [comment]
Sorry for the delay @LiorRabin ! Got it to work well with this configuration. The main issue is that when running tests, only currency.js passes (others try to bind to 8081 - already in use by currency.js). I think this doens't have to do with the container setup.

NOTE: the mongo endpoint is now mongodb://mongo/... instead of localhost.

Do you want me to document the setup?

**Comment by LiorRabin** [comment]
@JeremiGendron hello :) I will review it during today and let you know if I have any questions/comments. Thank you.

**Comment by JeremiGendron** [comment]
Ok let me do that for you.

**Comment by JeremiGendron** [comment]
@LiorRabin I've worked out a good solution, now I'm optimizing it so that it isn't necessary to node-gyp build every container (shared image for prod, qa, test but different container params). Will push changes.

**Comment by LiorRabin** [comment]
@JeremiGendron great! Looking forward to it :)

**Comment by JeremiGendron** [comment]
@LiorRabin I'll wrap it up tomorrow

**Comment by JeremiGendron** [comment]
@LiorRabin Review this again, some details in README :clinking_glasses:

**Comment by LiorRabin** [comment]
@JeremiGendron hello.
I'm so sorry but this is not our intention.
I would like to be able to inject ENV variables into the docker (for example the port to listen to which is hardcoded in the way that you wrote it).
Also, I don't need a different file for each docker.
If you look at the code you will see a config folder with files LOCAL,QA,PROD inside of it.
All of the variables in those files can be able to be over-rid using ENV variables and the file the server should run according to is the ENV variable called `ENV` as you can see in the `package.json` scripts.
Maybe this makes the request of the docker capabilities clearer for you.
Sorry for going back and forth on this, just need to have a solution suited for our use case.

**Comment by JeremiGendron** [comment]
Please be very clear, is the following ok?
`sudo docker run -v .:/im -p "XXXX:8080" -e "ENV=test" .`

Also, what do you want it to do? You are able to pass in the command you'd like to run through -e too.

**Comment by JeremiGendron** [comment]
docker-compose is not meant for that much granularity as far as I can tell, also the ports can be changed in the compose file as well as the command to_execute. The job would be done with a simple Dockerfile.

**Comment by LiorRabin** [comment]
Hi @JeremiGendron, please take a look at this:
https://github.com/ColuLocalNetwork/CLN-community-app/blob/master/docker-compose.yml
Maybe this will clear up what I'm trying to achieve...

**Comment by JeremiGendron** [comment]
`cp .env.dist .env` so you want multiple .env files? I can do that.

**Comment by LiorRabin** [comment]
https://github.com/ColuLocalNetwork/CLN-community-app/blob/ada2cf213f1d23e4651fdd5955551ffa24d606fb/docker-compose.yml#L3

https://github.com/ColuLocalNetwork/CLN-community-app/blob/ada2cf213f1d23e4651fdd5955551ffa24d606fb/docker-compose.yml#L39

Those are the sections we need, working in the same logic with the ability to see env variables, without all the .env files

**Comment by JeremiGendron** [comment]
The env variables work due to the env files being loaded first. Otherwise there's no way to have $PORT (env_file)

**Comment by LiorRabin** [comment]
yes, I understand this

**Comment by pgrzesik** [comment]
hey @LiorRabin - are you looking for another person to try taking a stab at it ? I can do it in my free time :)

**Comment by LiorRabin** [comment]
Hey @pgrzeski, we will maybe open a new bounty at a later stage so stay tuned

**Pull request closed without merging**